assets/js/components/Collection/Form.jsx

```diff
@@ -83,64 +57,53 @@ const CollectionForm = ({ collection }) => {
     history.push("/collection/list");
   };
 
-  const handleInputChange = e => {
-    setFormValues({
-      ...formValues,
-      [e.target.name]: e.target.value
-    });
-  };
-
-  const handleIsFeaturedChange = e => {
-    const prevVal = formValues.featured;
-    setFormValues({
-      ...formValues,
-      featured: !prevVal
-    });
-  };
-
-  const handleSubmit = e => {
-    e.preventDefault();
-
-    // Convert keywords to an array as defined in GraphQL
-    let values = { ...formValues };
-    values.keywords = formValues.keywords
-      .split(",")
-      .map(keyword => keyword.trim());
-
+  const onSubmit = data => {
     if (!collection) {
       createCollection({
-        variables: { ...values }
+        variables: { ...data }
       });
     } else {
       updateCollection({
-        variables: { ...values, collectionId: collection.id }
+        variables: { ...data, collectionId: collection.id }
       });
     }
   };
 
   return (
     <div>
-      <form onSubmit={handleSubmit} data-testid="collection-form">
+      <form onSubmit={handleSubmit(onSubmit)} data-testid="collection-form">
         <div className="columns is-centered">
           <div className="column is-half">
-            <UIFormInput
-              placeholder="Add collection Name"
-              type="text"
-              name="collectionName"
-              id="collectionName"
-              label="Collection Name"
-              value={formValues.collectionName || ""}
-              onChange={handleInputChange}
-              data-testid="collection-name"
-            />
+            <div className="field">
+              <label htmlFor="collectionName" className="label">
+                Collection Name
+              </label>
+              <div className="control">
+                <input
+                  placeholder="Add collection Name"
+                  className={`input ${errors.collectioName ? "is-danger" : ""}`}
+                  type="text"
+                  ref={register({ required: true })}
+                  name="collectionName"
+                  id="colletion-name"
+                  defaultValue={collection ? collection.name : ""}
+                  data-testid="collection-name"
```

Comment:
How about we establish a pattern to help us in advance as we start increasing test coverage.   For naming `data-testid`s, if it's a form element, let's use this convention as a rule of thumb:

`data-testid="[type of form element - name of form element`

So `data-testid="collection-name"` becomes `data-testid="input-collection-name"`

... and so forth in all the form elements which have test ids.

Thoughts?